### ISSUE TITLE
JavaScript error probably caused by Prototype.JS

### ISSUE THREAD

**Issue by tomnoir** [Really-Simple-Plugins/complianz-gdpr#411]
We have an issue on some pages with Complianz 7.0.7 premium.

It seems like that on all pages that use Prototype.JS, there is a JS error in the cmplz_set_category_as_body_class function. It iterates over the cookie categories and tries to add these as classes to the body tag. Yet it seems, that in this case it not only iterates over the array of string, but also built in methods of the JS array (like the .each-function). If you could add a check whether the current element in the iterator is a string, this would probably solve the problem. 

<img width="1136" alt="image" src="https://github.com/Really-Simple-Plugins/complianz-gdpr/assets/5177346/8037f7fb-9601-45b4-8254-4771ec5b19d7">

<img width="604" alt="image" src="https://github.com/Really-Simple-Plugins/complianz-gdpr/assets/5177346/13a138e3-33fc-4b1b-8487-8d67283501a0">

Best, Thomas

**Comment by rlankhorst** [comment]
@tomnoir Thanks for reporting the issue. In regular foreach loops I always add a .hasOwnproperty check, which catches this kind of stuff, but in this case your solution works fine. I have created a branch in premium and free. We'll include it in the next update. In the meantime, you can copy over the complianz.min.js and complianz.js files from cookiebanner/js in this branch from the free plugin:

https://github.com/Really-Simple-Plugins/complianz-gdpr/tree/fix-non-string-in-set_category_as_body_class

Let me know if this resolves your issue!

**Comment by tomnoir** [comment]
@rlankhorst Yes, this fixes it and the banner works. Thank you for the quick reply and fix!

**Comment by rlankhorst** [comment]
@tomnoir great, thanks for confirming! I'll close this issue, it will get merged in one of the coming releases.
